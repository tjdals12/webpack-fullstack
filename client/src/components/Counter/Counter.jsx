import React from 'react';

class Counter extends React.Component {
    constructor(props) {
        super(props);
        this.state = { count: 0 };
    }

    onIncrease = () => {
        this.setState(prevState => {
            return {
                ...prevState,
                count: prevState.count + 1,
            };
        });
    };

    onDecrease = () => {
        this.setState(prevState => {
            return {
                ...prevState,
                count: prevState.count - 1,
            };
        });
    };

    render() {
        const { count } = this.state;
        const { onIncrease, onDecrease } = this;

        return (
            <div>
                <p>{count}</p>
                <button type="button" onClick={() => onIncrease()}>
                    +
                </button>
                <button type="button" onClick={() => onDecrease()}>
                    -
                </button>
            </div>
        );
    }
}

export default Counter;
