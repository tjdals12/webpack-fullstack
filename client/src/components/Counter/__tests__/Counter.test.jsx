import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Counter from '../Counter';

describe('<Counter />', () => {
    it('matches snapshot', () => {
        const utils = render(<Counter />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has a count and two buttons', () => {
        const utils = render(<Counter />);

        utils.getByText('0');
        utils.getByText('+');
        utils.getByText('-');
    });

    it('onIncrease', () => {
        const utils = render(<Counter />);

        const count = utils.getByText('0');
        const plusButton = utils.getByText('+');

        fireEvent.click(plusButton);
        expect(count.textContent).toBe('1');
    });

    it('onDecrease', () => {
        const utils = render(<Counter />);

        const count = utils.getByText('0');
        const minusButton = utils.getByText('-');

        fireEvent.click(minusButton);
        expect(count.textContent).toBe('-1');
    });
});
