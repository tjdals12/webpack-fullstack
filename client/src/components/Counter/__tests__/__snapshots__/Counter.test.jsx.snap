// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot 1`] = `
<div>
  <div>
    <p>
      0
    </p>
    <button
      type="button"
    >
      +
    </button>
    <button
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
